instance/models/openedx_instance.py

```diff
@@ -389,3 +394,97 @@ def delete(self, *args, **kwargs):
         self.deprovision_s3()
         self.deprovision_rabbitmq()
         super().delete(*args, **kwargs)
+
+    @property
+    def consul_prefix(self):
+        """
+        This is a property that helps determining a specific instance's rpefix in
+        Consul. The prefix helps in keeping different instances' configurations
+        separate from overridings and modifications.
+
+        :return: A unique Key-Value prefix for this instance in Consul.
+        """
+        return '{ocim}/instances/{instance}/'.format(ocim=settings.OCIM_ID, instance=self.id)
+
+    def _generate_consul_metadata(self):
+        """
+        Collects required configurations for this instance to reflect on Consul.
+        You can add, delete or alter configurations keys and values here which are
+        going to ve reflected in Consul immediately.
+
+        :return: A dict of the configurations.
+        """
+        active_servers = self.get_active_appservers()
+        basic_auth = self.http_auth_info_base64()
+        enable_health_checks = int(active_servers.count() > 1)
+        active_servers_data = list(active_servers.annotate(public_ip=F('server___public_ip')).values('id', 'public_ip'))
+
+        configurations = {
+            'name': self.name,
+            'domains': self.get_load_balanced_domains(),
+            'health_checks_enabled': enable_health_checks,
+            'basic_auth': basic_auth.decode('utf-8'),
```

Comment:
Just as a side note, the UTF-8 encoding is the default in Python 3, so just `decode()` will do the trick.  In some cases it may be desirable to be explicit about the encoding anyway, but in this particular case the encoding is rather irrelevant.